src/main/scala/org/constellation/storage/SnapshotService.scala

```diff
@@ -316,8 +327,15 @@ class SnapshotService[F[_]: Concurrent](
 
       _ <- checkpointService.applySnapshot(currentSnapshot.checkpointBlocks.toList)
       _ <- dao.metrics.updateMetricAsync(Metrics.lastSnapshotHash, currentSnapshot.hash)
+      _ <- dao.metrics.incrementMetricAsync(Metrics.snapshotCount)
     } yield ()
 
+    val toEither: F[Either[SnapshotError, Unit]] = applyAfter.map(_ => Right(()))
+    toEither.handleError(
+      e => Left(SnapshotUnexpectedError(e))
+    )
```

Comment:
If `F[_]: Sync` implements a MonadError, then:

```scala
applyAfter.attempt
   .map(either => either
        .leftMap(e => SnapshotUnexpectedError(e))
         .map(_ => ())
   )
```
or with monad transformer:

```scala
EitherT(applyAfter.attempt)
  .leftMap(e => SnapshotUnexpectedError(e))
  .map(_ => ())
```

I mean we have abstractions provided by cats to avoid converting things between each other manually. The problem is that IDE sometimes does not assist with importing `cats.syntax` or `cats.implicits`

Reply:
@kpudlik thanks ! :muscle: